dali/ft/filecopy.cpp

```diff
@@ -1501,6 +1503,13 @@ void FileSprayer::analyseFileHeaders(bool setcurheadersize)
         {
             try
             {
+                if (distributedSource)
+                {
+                    Owned<IDistributedFilePart> curPart = distributedSource->getPart(idx);
+                    IPropertyTree& curProps = curPart->queryAttributes();
+                    cur.xmlHeaderLength = curProps.getPropInt(FAheaderLength, 0);
```

Comment:
Are these values immediately being rewritten by either locateContentHeader or srcFormat.headerLength, in the next few lines?

I think what you want would be 1. if the values are in file level DFS use it.  2. or if the values are in PART level DFS, use it.  3. otherwise locateContentHeader.